Modules/Microsoft365DSC/DSCResources/MSFT_IntuneDeviceConfigurationPolicyWindows10/MSFT_IntuneDeviceConfigurationPolicyWindows10.schema.mof

```diff
@@ -1,10 +1,11 @@
 [ClassVersion("1.0.0.0")]
-class MSFT_DeviceManagementConfigurationPolicyAssignments
+class MSFT_IntuneDeviceConfigurationPolicyWindows10Assignments
```

Comment:
This would result in a breaking change.

Reply:
Is this change really needed? The CIM Instance is embedded into this resource. Are there any other CIM Instances with the previous name somewhere else (and do they have the same properties?

Reply:
Basically almost all Intune resources that support Assignments use exactly the same CIM instance *MSFT_DeviceManagementConfigurationPolicyAssignments*, which means that in order to support the changes I made in this PR without changing the CIM instance I would need to change all Intune resources at the same time otherwise it gives error messages when importing the module.

I discussed this privately with Nik several months ago.

Reply:
Do all other resources need the display name parameter? If yes, we should change it globally.

Reply:
I am fine with this PR starting by addressing one resource, and then later submit a larger PR to address them all. However, we need to confirm whether or not we really need to rename the CIMInstance right now for this to work or if this can be done in April. I can hold today's release for a few hours, but if we can't resolve the renaming question soon, this will need to get pushed to next week at a minimum. Thanks

Reply:
I could add the parameter to the schema of all of them without renaming the CIM instance sure, but without adding the remaining code I've added to this PR, that would take me ages but also because several of those resources have been created manually instead of the DRG so their surrounding code is totally different.

Can I do that? I don't mind making those changes in the next couple of days, I can wait for next week's release, I have no rush to get this in.

Reply:
Thanks. What I am suggesting for now is to keep the additional code in, but keep the current CIMInstance name. Then later submit two other PRs. The first one to add the data type to all resources, which could get released anytime when ready, then a second one to rename the CIMInstance which would get release as breaking changes in April. Would that be feasible?

Reply:
groupDisplayName, not dataType :) But yeah that's what I mentioned keep the code I added, change CIMInstance name back to the original but then I'll need to add groupDisplayName property to all resources in this same PR otherwise it won't work, all resources must be changed at the same time, then the code to make them work could be added later on.

That being said, do you still prefer to have individual CIMInstance names per resource (making those changes in April) or keep the same name for all of them?

Reply:
Coming from a standpoint, that we should reuse as much code as possible, and if there is no eminent reason to change the CIM Instance name, I would stick one name across all resources.